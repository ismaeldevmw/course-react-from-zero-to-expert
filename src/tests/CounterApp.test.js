import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import CounterApp from '../CounterApp';

describe('Tests into <CounterApp />', () => {
    
    test('should show <CounterApp /> alright', () => {
        
        const wrapper = shallow(<CounterApp />);
        expect(wrapper).toMatchSnapshot();

    });

    test('should show default value 100', () => {
        
        const wrapper = shallow( <CounterApp value={ 100 }/> );
        const counterText = wrapper.find('h2').text().trim();

        expect( counterText ).toBe('100');
    });

});