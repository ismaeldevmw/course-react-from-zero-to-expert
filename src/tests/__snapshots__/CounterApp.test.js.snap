// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests into <CounterApp /> should show <CounterApp /> alright 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
